bridget/eddy/footprint.py

```diff
@@ -240,108 +241,190 @@ def obukhov_length_calc(rho, c_p, u_star, t_air, hts, k = 0.4):
 
 
 
-def footprint(t_air, a, p, u, cov_uw, cov_vw, cov_wt, var_v, z = 2.0, method = "Kormann"):
+def footprint(time, t_air, a, p, u, cov_uw, cov_vw, cov_wt, var_v, direction, tstamp, z = 2.0, fetch = 200, grid = 500, method = "Kormann"):
+    """
+    Top-level footprint function.
+
+    The calculation method is selected with the method parameter.
+    Currently implemented methods are:
+        - method = "Kormann": Kormann and Meixner (2001)
+
+    Other methods will follow, with the highest priority for the Kljun model (2015)
+
+    For further information about the methods check the References section.
+    Leclerc and Foken (2014) are giving an overview of avaiable footprint models in chapter 1.3 and table 1.3 (p.12)
+
+    Parameters
+    ----------
+    time : list, datetime
+        datetime list corresponding to the other variables, format: '%m.%d.%Y %H:%M'.
+    t_air : list, float
+        air temperature [°C].
+    a : list, float
+        absolute humidity [g/m3].
+    p : list, float
+        air pressure [hPa].
+    u : list, float
+        horizontal wind component for the direction in which the sonic is oriented [m/s].
+    cov_uw : list, float
+        Covariance between the wind components u and w [m2/s2].
+    cov_vw : list, float
+        Covariance between the wind components v and w [m2/s2].
+    cov_wt : list, float
+        Covariance between the wind component w and temperature [(m*°C)/s].
+    var_v : list, float
+        variance of v (horizontal wind component for the direction rectangular to the orientation of the sonic) [m/s].
+    direction : list, float
+        wind direction [°].
+    tstamp : int
+        index of the line/time at which the footprint is to be calculated.
+    z : float
+        measurement height [m].
+    fetch : int
+        upwind distance over which calculation domain to extends [m].
+    grid : int
+        total calculation grid.
+    method : string
+        pick a calculation method.
+
+    References
+    ----------
+    Leclerc MY, Foken T (eds) (2014) Footprints in Micrometeorology and Ecology. Springer, Berlin
+    Kormann & Meixner 2001, An Analytical Footprint Model For Non-Neutral Stratification
+
+    """
     if method.lower() == 'kormann':
-      return _footprint_korman(t_air, a, p, u, cov_uw, cov_vw, cov_wt, var_v, z)
-    elif method.lower() == 'other_guy':
-      pass
+      return _footprint_kormann(time, t_air, a, p, u, cov_uw, cov_vw, cov_wt, var_v, direction, tstamp, z, fetch, grid)
+    else:
+      raise NotImplementedError('Hold tight, other methods are about to come')
 
 
 
 
-def _footprint_korman(t_air, a, p, u, cov_uw, cov_vw, cov_wt, var_v, z = 2.0):
+def _footprint_kormann(time, t_air, a, p, u, cov_uw, cov_vw, cov_wt, var_v, direction, tstamp, z, fetch, grid):
```

Comment:
@AlexDo1 , one more minor, but very important thing: `time` is one the the built-in libraries. Please do not use that name for a function parameter. It could also be misleading, as we expect a `datetime` object here (or a list thereof). I would suggest to use something as `index`, `dt_index`, `datetime_index` or `dates`

Reply:
I renamed the time parameter to dt_index in the next commit.